partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -487,10 +487,77 @@ internal object RexConverter {
             // Args
             val args = node.args.map { visitExprCoerce(it, context) }
             // Rex
+            if (node.setq != null) {
+                if (isCollAgg(node)) {
+                    return callToCollAgg(id, node.setq!!, args)
+                } else {
+                    error("Currently, only COLL_<AGG> may use set quantifiers.")
+                }
+            }
             val op = rexOpCallUnresolved(id, args)
             return rex(type, op)
         }
 
+        /**
+         * @return whether call is `COLL_<AGG>`.
+         */
+        private fun isCollAgg(node: Expr.Call): Boolean {
+            val id = node.function as? org.partiql.ast.Identifier.Symbol ?: return false
+            return id.symbol.lowercase().startsWith("coll_")
+        }
+
+        /**
+         * Converts inputs to `COLL_<AGG>` when DISTINCT is used.
```

Comment:
With this relatively involved rewrite of `COLL_<AGG>` fns, I wonder if we should model them as separate special form nodes within the plan (and maybe AST?). `COLL_<AGG>`s are `Rex.Op`s so it feels overcomplicated to have a `Rex.Op.Select` with a `Rel.Op.Distinct` and `Rel.Op.Scan` nested within.

Reply:
Coming from my recent experience working with the SQL set ops (https://github.com/partiql/partiql-lang-kotlin/pull/1506), I found it hard to work with and debug plans with these additional nested nodes.

Reply:
ACK. From @RCHowell's recent discussion, I've added a distinct variant to each function.